### DISCUSSION TITLE
RobotCreator/RobotDescriptor

### DISCUSSION THREAD

**Discussion by maidenone** [Jsevenh/RobotDescriptor#8] [General]
I have started to look at your code and its way better then mine. You created a pull request in RobotCreator that i merged into develop, but that removed all my work, but that is fine. I can migrate it to your new code base. Do you want to change the name to Robot Descriptor? I can give you admin access to maidenone/RobotCreator if you want to continue there, was going to make some changes while reviewing your code and saw that you had made more recent changes in this repository.

Maybe we should proceed with a common name and repo?

**Comment by Jsevenh** [comment]
> was going to make some changes while reviewing your code and saw that you had made more recent changes in this repository.

do make the changes I've realized I write horrible code with a lot of repetitions with  little optimization ,If you happen to have inquiries regarding any ambiguous aspects within my code, feel free to reach out. I'm open to discussions and happy to provide clarification.

**Comment by Jsevenh** [comment]
> Do you want to change the name to Robot Descriptor? I can give you admin access to maidenone/RobotCreator if you want to continue there

I might pass on the admin access to maidenone/RobotCreator. GitHub wizardry isn't exactly my forte,  yep a change of name to RobotDescriptor would be awesome considering  all that is being done is generating  a robot description format and proceeding with a common repo is a great Idea

**Reply by maidenone** [reply]
In that case, i think i revert your pullrequest, and fork your repository and create pullrequests on your repo.

"normal" git workflow is that you have a master branch for stable releases, a develop branch foe new features, and each developer creates a "feature" branch for what they are currently working on. feature branches go into develop, and once develop is stable enough it is merged into master and a release is made. normally master and develop is locked and can not be pushed to directly.

**Comment by Jsevenh** [comment]
I think proceeding from your repo might be a good idea , with a suggestion to change the name to RobotDescriptor

**Reply by maidenone** [reply]
we are in sync ^_^

**Reply by Jsevenh** [reply]
perhaps we can proceed from your repo then

**Reply by maidenone** [reply]
[[](url)](https://github.com/maidenone/RobotDescriptor)

**Reply by Jsevenh** [reply]
awesome

**Comment by Jsevenh** [comment]
So basically I'm done with  the first part, that is creating the world widget ,the next part is model imports, I suppose starting a discussing on your repo  about that will be great

**Reply by maidenone** [reply]
https://github.com/maidenone/RobotDescriptor/discussions/12

**Comment by maidenone** [comment]
Created a test issue / pull request to verify workflow. 
see:
https://github.com/maidenone/RobotDescriptor/issues/13
and review:
https://github.com/maidenone/RobotDescriptor/pull/14

You are added as "contributor" so you should be able to work as "normally" on my branch now. You can not push directly to develop, neither can i. you have to follow the workflow in:
https://github.com/maidenone/RobotDescriptor/discussions/12
